Userland/Demos/Mandelbrot/Mandelbrot.cpp

```diff
@@ -106,8 +106,11 @@ class MandelbrotSet {
         if (iteration == max_iterations)
             return iteration;
 
-        auto lz = sqrt(x * x + y * y) / 2;
```

Comment:
Also, I'm not sure this is a mistake. The "before" screenshots are a pretty common look of Mandelbrot viewers.

Reply:
> Also, I'm not sure this is a mistake. The "before" screenshots are a pretty common look of Mandelbrot viewers.

The linked Wikipedia article in the comment by the original author has
pseudocode for the smooth coloring rendering:

```
    while x*x + y*y ≤ (1 << 16) and iteration < max_iteration do
        xtemp:= x*x - y*y + x0
        y:= 2*x*y + y0
        x:= xtemp
        iteration:= iteration + 1

    // Used to avoid floating point issues with points inside the set.
    if iteration < max_iteration then
        // sqrt of inner term removed using log simplification rules.
        log_zn:= log(x*x + y*y) / 2
        nu:= log(log_zn / log(2)) / log(2)
        // Rearranging the potential function.
        // Dividing log_zn by log(2) instead of log(N = 1<<8)
        // because we want the entire palette to range from the
        // center to radius 2, NOT our bailout radius.
        iteration:= iteration + 1 - nu
```

Which uses `log()` instead of `sqrt()`. Also note that `nu` should be subtracted
not added.

The bands of color look (escape time algorithm) is just returning `iteration`
after the while loop (no need to calculate the fractional part).

Reply:
The correct formula is this (from https://linas.org/art-gallery/escape/escape.html)
>      mu = m + frac = n + 1 - log (log  |Z(n)|) / log 2

But the wiki pseudocode divides by `log(2)` one more time for some reason.

I am not fully satisfied with the smooth coloring render right now; the radius=2 circle outline is still visible which most of the renders elsewhere don't have. I am trying to debug that.

Reply:
> the radius=2 circle outline is still visible which most of the renders elsewhere don't have.

Increasing the escape radius gives perfect smooth coloring. So no bug in the implementation.

Reply:
Right, my point was that the existing look is common too, probably because it reveals a lot more structure. Thanks for updating the commit description!